src/compiler/checker.ts

```diff
@@ -19140,6 +19177,12 @@ namespace ts {
                     case SyntaxKind.PropertyAssignment:
                         // We might be in `a = { b: this.b }`, so keep looking. See `tests/cases/compiler/useBeforeDeclaration_propertyAssignment.ts`.
                         return false;
+                    case SyntaxKind.ExpressionWithTypeArguments:
+                    case SyntaxKind.HeritageClause:
+                    case SyntaxKind.ComputedPropertyName:
+                    case SyntaxKind.GetAccessor:
+                    case SyntaxKind.SetAccessor:
+                        return false;

```

Comment:
actually, where is this called? I think it's part of `isInPropertyInitializer`, but I don't see any new calls to it, and don't see how the call in `checkPropertyNotUsedBeforeDeclaration` changes with the change in `isBlockScopedNameDeclaredBeforeUse`.

Reply:
@sandersn it's been a while since diving into this code but IIRC this was necessary to have the compiler check some node edge cases it previously didn't need to. For example, for this class property:

```ts
withinObjectLiteralGetterName: any = {
    get [this.withinObjectLiteralGetterName]() {
        return true;
    }
};
```

50% as a response & 50% for my own notes:

* `isInPropertyInitializer` is used by `checkPropertyNotUsedBeforeDeclaration` to tell if the node being checked is within its a property initializer
* `checkPropertyNotUsedBeforeDeclaration` is used by `checkPropertyAccessExpressionOrQualifiedName` as one case to check the node on, if its `prop` (symbol) exists (per `getPropertyOfType`)
* `checkPropertyAccessExpressionOrQualifiedName` is called by `checkPropertyAccessExpression`, which is called by a few other places

Reply:
Thanks, that call-graph analysis is exactly what I needed.